lib/lumberjack/beats/server.rb

```diff
@@ -139,13 +130,80 @@ def close
       @close.make_true
       @server.close unless @server.closed?
     end
-  end # class Server
+
+    private
+    def retrieve_current_port
+      # Query the port in case the port number is '0'
+      # TCPServer#addr == [ address_family, port, address, address ]
+      @server.addr[1]
+    end
+
+    def certificate_authorities
+      Array(@options[:ssl_certificate_authorities])
+    end
+
+    def server_private_key
+      OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
```

Comment:
Good point I think the validation in the `#initialize` is wrong I will reflect to make the following change.

When `ssl` is set to true:
- **ssl_key** need to be set.
- **ssl_certificate** need to be set.

When `ssl_verify_mode` is set to other than `none`:
- **ssl_certificate_authorities** need to be set.

Its doesnt make sense to me to be able to remove the certificate and the key if we turn on ssl.